src/buffer.h

```diff
@@ -45,17 +45,37 @@ struct mpd_buffer {
 	unsigned read;
 
 	/** the actual buffer */
-	unsigned char data[4096];
+	unsigned char *data;
+
+	/** size of buffer */
+	size_t data_len;
 };
 
 /**
  * Initialize an empty buffer.
  */
-static inline void
+static inline void *
```

Comment:
Undocumented return value. Why would a caller ever be interested in the value of the allocated buffer pointer?

Reply:
This can be changed to bool. The idea is to check if the buffer was able to be allocated

Reply:
If that's your idea, then your return value should express exactly that. And should be documented.